tst/EnergyPlus/unit/OccupantDiversity.unit.cc

```diff
@@ -0,0 +1,2063 @@
+        "  For: AllDays,            !- Field 4",
+        "  Until: 24:00,1.0;        !- Field 5",
+        " ",
+        "Schedule:Compact,",
+        "  EquipSched,              !- Name",
+        "  Any Number,              !- Schedule Type Limits Name",
+        "  Through: 12/31,          !- Field 3",
+        "  For: WinterDesignDay,    !- Field 4",
+        "  Until: 24:00,0.0,        !- Field 5",
+        "  For: AllOtherDays,       !- Field 4",
+        "  Until: 24:00,1.0;        !- Field 5",
+        " ",
+        "Schedule:Compact,",
+        "  ActivityLevelSched,      !- Name",
+        "  Any Number,              !- Schedule Type Limits Name",
+        "  Through: 12/31,          !- Field 3",
+        "  For: AllDays,            !- Field 4",
+        "  Until: 24:00,120.0;      !- Field 5",
+        " ",
+        "Schedule:Compact,",
+        "  FanOpModeSchedule,       !- Name",
+        "  Any Number,              !- Schedule Type Limits Name",
+        "  Through: 12/31,          !- Field 1",
+        "  For: AllDays,            !- Field 2",
+        "  Until: 24:00,1.0;        !- Field 7",
+        " ",
+        "Curve:Biquadratic,",
+        "  Biquadratic,             !- Name",
+        "  1.0,                     !- Coefficient1 Constant",
+        "  0.0,                     !- Coefficient2 x",
+        "  0.0,                     !- Coefficient3 x**2",
+        "  0.0,                     !- Coefficient4 y",
+        "  0.0,                     !- Coefficient5 y**2",
+        "  0.0,                     !- Coefficient6 x*y",
+        "  5,                       !- Minimum Value of x",
+        "  40,                      !- Maximum Value of x",
+        "  5,                       !- Minimum Value of y",
+        "  40,                      !- Maximum Value of y",
+        "  ,                        !- Minimum Curve Output",
+        "  ,                        !- Maximum Curve Output",
+        "  Temperature,             !- Input Unit Type for X",
+        "  Temperature,             !- Input Unit Type for Y",
+        "  Dimensionless;           !- Output Unit Type",
+        " ",
+        "Curve:Cubic,",
+        "  Cubic,                   !- Name",
+        "  1.0,                     !- Coefficient1 Constant",
+        "  0.0,                     !- Coefficient2 x",
+        "  0.0,                     !- Coefficient3 x**2",
+        "  0,                       !- Coefficient4 x**3",
+        "  5,                       !- Minimum Value of x",
+        "  40,                      !- Maximum Value of x",
+        "  ,                        !- Minimum Curve Output",
+        "  ,                        !- Maximum Curve Output",
+        "  Temperature,             !- Input Unit Type for X",
+        "  Temperature;             !- Output Unit Type",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    ManageSimulation(*state);
+    EXPECT_EQ(1.0, state->dataSize->DBySys(1));
```

Comment:
Check that the Occupant Diversity calculated using the user-specified schedule is used for the calculations.